pwnlib/util/misc.py

```diff
@@ -367,13 +367,13 @@ def run_in_new_terminal(command, terminal=None, args=None, kill_at_exit=True, pr
                 with open('/proc/sys/kernel/osrelease', 'rb') as f:
                     is_wsl = b'icrosoft' in f.read()
             if is_wsl and which('cmd.exe') and which('wsl.exe') and which('bash.exe'):
-                terminal    = 'cmd.exe'
-                args        = ['/c', 'start']
+                terminal    = 'wt.exe'
```

Comment:
We cannot use wt.exe if the user does not have it installed. Also, the problem is not cmd.exe but `start` (which probably can take an extra cwd argument? not sure, I personally do not have any Windows machine thankfully, but you can run `start /?` in cmd to get some idea of whether it is possible). Otherwise, we could run wt.exe without start and the native terminal with start.

Reply:
maybe we can use `powershell` (yes , it can prase the file path of WSL ), but it will cost much more resources than `cmd`.

Reply:
Ah, I think I have a way to deal with `cmd`:

```python
        else:
            is_wsl = False
            if os.path.exists('/proc/sys/kernel/osrelease'):
                with open('/proc/sys/kernel/osrelease', 'rb') as f:
                    is_wsl = b'icrosoft' in f.read()
            if is_wsl and which('powershell.exe') and which('wsl.exe') and which('bash.exe'):
                terminal    = 'cmd.exe'
                args        = ['/c', 'start']
                distro_name = os.getenv('WSL_DISTRO_NAME')
                current_dir = os.getcwd()

                # Split pane in Windows Terminal
                if 'WT_SESSION' in os.environ and which('wt.exe'):
                    args.extend(['wt.exe', '-w', '0', 'split-pane'])
                if distro_name:
                    args.extend(['wsl.exe', '-d', distro_name, '--cd', current_dir, 'bash', '-c'])
                else:
                    args.extend(['bash.exe', '-c'])
```

![image](https://github.com/user-attachments/assets/761439be-659e-4f11-a25f-53c3cd006e1a)